spec/system.spec.js

```diff
@@ -347,21 +380,36 @@ describe('corp-semantic-release', function() {
   });
 
 
-  it('should only run post-success script when the git push command is successful', function() {
+  it('should run post-success script when the git push command is successful', function() {
     commitFeat();
-    const out = semanticRelease(`--post-success do-publish -v`);
+    const result = semanticRelease(`--post-success "npm run do-publish" -v`);
```

Comment:
Hi, sure. I've added an option to the whole command that tells it to mock the `git push` step and return specific code. Then I've added it to `semanticRelease`, so that it succeeds. Then if a test wants it to fail it can call `semanticRelease` with `--mock-push 1`, as options added later override those added earlier.

This is of course a very hacky solution, but I don't see any other possibility without more refactoring (see #41), which would be out of scope of this change.